geospatial/src/Dem.cc

```diff
@@ -275,35 +284,78 @@ float Dem::MaxElevation() const
   {
-    gzerr << "Can not retrieve WGS84 coordinates from non-Earth DEM."
+    gzerr << "Can not retrieve coordinates from unknown DEM."
             << std::endl;
     return false;
   }
 
   double geoTransf[6];
   if (this->dataPtr->dataSet->GetGeoTransform(geoTransf) == CE_None)
   {
-    OGRSpatialReference sourceCs;
-    OGRSpatialReference targetCs;
-    OGRCoordinateTransformation *cT;
+    OGRCoordinateTransformation *cT = nullptr;
     double xGeoDeg, yGeoDeg;
 
-    // Transform the terrain's coordinate system to WGS84
-    const char *importString
-        = strdup(this->dataPtr->dataSet->GetProjectionRef());
-    if (importString == nullptr || importString[0] == '\0')
+    if (this->dataPtr->sphericalCoordinates.Surface() ==
+        math::SphericalCoordinates::EARTH_WGS84)
     {
-      gzdbg << "Projection coordinate system undefined." << std::endl;
-      return false;
+      OGRSpatialReference sourceCs;
+      OGRSpatialReference targetCs;
+      // Transform the terrain's coordinate system to WGS84
+      const char *importString
+          = strdup(this->dataPtr->dataSet->GetProjectionRef());
+      if (importString == nullptr || importString[0] == '\0')
+      {
+        gzdbg << "Projection coordinate system undefined." << std::endl;
+        return false;
+      }
+      sourceCs.importFromWkt(&importString);
+      targetCs.SetWellKnownGeogCS("WGS84");
+      cT = OGRCreateCoordinateTransformation(&sourceCs, &targetCs);
+    }
+    else if (this->dataPtr->sphericalCoordinates.Surface() ==
+        math::SphericalCoordinates::MOON_SCS)
+    {
+      auto sourceCs = this->dataPtr->dataSet->GetSpatialRef();
+      OGRSpatialReference targetCs = OGRSpatialReference();
+
+      const char* MOON_LAT_LON_PROJ_STR =
+        "+proj=latlong +a=1737400 +b=1737400";
+
+      targetCs.importFromProj4(MOON_LAT_LON_PROJ_STR);
+
+      cT = OGRCreateCoordinateTransformation(
+          sourceCs, &targetCs);
+    }
+    else if (this->dataPtr->sphericalCoordinates.Surface() ==
+        math::SphericalCoordinates::CUSTOM_SURFACE)
+    {
+      auto sourceCs = this->dataPtr->dataSet->GetSpatialRef();
+      OGRSpatialReference targetCs = OGRSpatialReference();
+
+      double axisEquatorial =
+        this->dataPtr->sphericalCoordinates.SurfaceAxisEquatorial();
+      double axisPolar =
+        this->dataPtr->sphericalCoordinates.SurfaceAxisPolar();
+
+      std::string CUSTOM_SURFACE_LAT_LON_PROJ_STR =
+        "+proj=latlong +a=" + std::to_string(axisEquatorial) +
+        " +b=" + std::to_string(axisPolar);
+
+      targetCs.importFromProj4(CUSTOM_SURFACE_LAT_LON_PROJ_STR.c_str());
+
+      cT = OGRCreateCoordinateTransformation(
+          sourceCs, &targetCs);
     }
-    sourceCs.importFromWkt(&importString);
-    targetCs.SetWellKnownGeogCS("WGS84");
-    cT = OGRCreateCoordinateTransformation(&sourceCs, &targetCs);
+    else
+    {
+      gzerr << "Unsupported spherical coordinates found" << std::endl;
+    }
```

Comment:
I think this section can be removed since it will go to the next `if` statement (i.e., `if (nullptr == cT)`)

Reply:
Done https://github.com/gazebosim/gz-common/pull/377/commits/e42c08d9e12ccf1902a20f930322f79a89d03238